core-addon/Core.js

```diff
@@ -755,6 +755,37 @@ export default class Core {
       return s;
     }).filter(study => (!study.studyPaused || study.studyJoined));
 
+    // Only show study if it is compatible with this version of the core add-on.
+    const coreVersion = browser.runtime.getManifest().version;
+
+    // Match the version number used by the core add-on, such as "1.3.7buildid20220107.052711".
+    // this uses capture groups to isolate the part before the build ID, the semantic version number "1.3.7".
+    const versionRegex = /^([0-9]+)\.([0-9]+)\.([0-9])s*/;
```

Comment:
what's with the `s*`?

Reply:
> what's with the `s*`?

The signing infra sets a long build ID as part of the version string, like: `1.3.7buildid20220107.052711`
I didn't want to have to worry about how long it might be, so I wanted to be on the safe side.

Any suggestions?

Reply:
I'll make it `\\s*` otherwise